docs/source/daml2ts/index.rst

```diff
@@ -0,0 +1,291 @@
+.. Copyright (c) 2020 The DAML Authors. All rights reserved.
+.. SPDX-License-Identifier: Apache-2.0
+
+Typescript!
+###########
+
+daml2ts : Typescript of DAML
+============================
+
+Introduction
+------------
+``daml2ts`` generates `json-api <../json-api/index.html>`_ compatible Typescript code. It accepts compiled DAML modules as input. The Typescript source files it produces contain type, value and function definitions that model the DAML entities it finds.
+
+The code that ``daml2ts`` generates is expressed in terms of the packages `@digital-asset/daml-types <https://github.com/digital-asset/daml/tree/master/language-support/ts/daml-types>`_ and `@mojotech/json-type-validation <https://github.com/mojotech/json-type-validation>`_ packages. Accordingly, every Typescript file generated begins with the following snippet.
+
+.. code-block:: typescript
+   :linenos:
+
+   /* eslint-disable @typescript-eslint/camelcase */
+   /* eslint-disable @typescript-eslint/no-use-before-define */
+   import * as jtv from '@mojotech/json-type-validation';
+   import * as daml from '@digitalasset/daml-types';
+
+@digitalasset/daml-types primitives
+------------------------------------
+
+To understand the Typescript code generated by ``daml2ts``, it is helpful to keep in mind this quick review of the primitives provided by @digitalasset/daml-types.
+
+**Interfaces**:
+
+``Serializable<T>``
+  An object that extends the ``Serializable<T>`` interface has a field ``decoder: () => jtv.Decoder<T>``.
+``Template<T extends object, K = unknown>``
```

Comment:
Again, I don't think the actual structure of `Template` is important but rather it's purpose: It's an interface for companion objects that correspond to templates in DAML. Similar for `Choice`.

Reply:
✔️